tdrs-frontend/src/components/Reports/Reports.jsx

```diff
@@ -280,18 +295,53 @@ function Reports() {
               </select>
             </label>
           </div>
-          <Button className="margin-y-4" type="button" onClick={handleSearch}>
+          <Button
+            className="margin-y-4"
+            type="button"
+            onClick={() => {
+              if (uploadedFiles.length > 0) {
+                setErrorModalVisible(true)
+              } else {
+                handleSearch()
+              }
+            }}
+          >
             Search
           </Button>
         </form>
       </div>
       {isUploadReportToggled && (
         <UploadReport
           stt={stt?.id}
-          header={submittedHeader}
+          header={`${currentStt} - Fiscal Year ${selectedYear} - ${quarters[selectedQuarter]}`}
           handleCancel={() => setIsToggled(false)}
         />
       )}
+      <Modal
+        title="Files Not Submitted"
+        message="Your files have not been submitted. Searching without submitting will discard your changes and remove any uploaded"
```

Comment:
Is the message complete? Maybe "...Uploaded FILE"?

Reply:
Nice catch!  Probably uploaded "file(s)" to also include the possibility of multiple non-submitted files.

Reply:
wow, not sure how that happened honestly. code goblins 😄